src/Staking.sol

```diff
@@ -196,45 +213,45 @@ contract StakeRegistry is AccessControl, Pausable {
     ////////////////////////////////////////
 
     /**
-     * @dev Checks to see if `overlay` is frozen.
-     * @param overlay Overlay of staked overlay
+     * @dev Checks to see if `address` is frozen.
+     * @param _owner owner of staked address
      *
      * Returns a boolean value indicating whether the operation succeeded.
      */
-    function overlayNotFrozen(bytes32 overlay) internal view returns (bool) {
-        return stakes[overlay].lastUpdatedBlockNumber < block.number;
+    function addressNotFrozen(address _owner) internal view returns (bool) {
```

Comment:
[CRITICAL] this will allow the same node address to run several (any number of) nodes on different overlays and act as if staked on all of them!!

If you do it this way, overlay should also be passed from the redis contract and checked against the current one  recorded in the stakes map.

Reply:
Copying this from private discussion

I dont think there is an attack vector. 
There is no case where I see that node can have 2 overlays staked
As we have multiple safe guards
There is only one overlay associated per address mapping at the time
Also when overlay changes, then we have lastUpdatedBlockNumber reset, which forces nodes out of the game for 2 rounds.
We have overlay checked on commits, so what nodes submit is checked per what is in staking mapping and needs to be the same, if not, the node cant submit or play. 
As we are using msg.sender it cant be manipulated to submit for someone else

Reply:
indeed no attack, but it has nothing to do with whats in the staking contract.

it is solely because this line in redis
https://github.com/ethersphere/storage-incentives/blob/25e7b18d6140a8290b905c2c76bbbaad94d23038/src/Redistribution.sol#L303
and this
https://github.com/ethersphere/storage-incentives/blob/25e7b18d6140a8290b905c2c76bbbaad94d23038/src/Redistribution.sol#L375